### PR TITLE
feat: added SignatureMediaTypes to RemoteVerifyOptions

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR adds SignatureMediaTypes to RemoteVerifyOptions.

The details regarding this PR is tracked in #217. 

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: notation.go [hunk 1]
```diff
@@ -256,7 +264,11 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, re
 			if err != nil {
 				return ErrorSignatureRetrievalFailed{Msg: fmt.Sprintf("unable to retrieve digital signature with digest %q associated with %q from the registry, error : %v", sigManifestDesc.Digest, artifactRef, err.Error())}
 			}
-
+			// checking the media type after fetching the blob is slow.
+			// created an issue to track this: https://github.com/notaryproject/notation-go/issues/215
+			if len(remoteOpts.SignatureMediaTypes) != 0 && !slices.Contains(remoteOpts.SignatureMediaTypes, sigDesc.MediaType) {
+				continue
```
Comment (shizhMSFT) [inline notation.go:270]: We need to add a log here.

Comment (patrickzheng200) [reply notation.go:271]: Code updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by priteshbandi** [comment]
Have added couple of comments on https://github.com/notaryproject/notation-go/issues/217, lets discuss in the issue about this feature.

**Comment by patrickzheng200** [comment]
Closed this PR as the issue discussion is still in progress.

**Pull request closed without merging**